src/vmm/src/rpc_interface.rs

```diff
@@ -0,0 +1,312 @@
+// Copyright 2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+use std::fmt::{Display, Formatter};
+use std::sync::{Arc, Mutex};
+
+use super::Vmm;
+
+use super::Error as VmmError;
+use builder::StartMicrovmError;
+use controller::VmmController;
+use polly::event_manager::EventManager;
+use resources::VmResources;
+use seccomp::BpfProgram;
+use vmm_config;
+use vmm_config::boot_source::{BootSourceConfig, BootSourceConfigError};
+use vmm_config::drive::{BlockDeviceConfig, DriveError};
+use vmm_config::logger::{LoggerConfig, LoggerConfigError};
+use vmm_config::machine_config::{VmConfig, VmConfigError};
+use vmm_config::net::{
+    NetworkInterfaceConfig, NetworkInterfaceError, NetworkInterfaceUpdateConfig,
+};
+use vmm_config::vsock::{VsockDeviceConfig, VsockError};
+
+/// This enum represents the public interface of the VMM. Each action contains various
+/// bits of information (ids, paths, etc.).
+#[derive(PartialEq)]
+pub enum VmmAction {
+    /// Configure the boot source of the microVM using as input the `ConfigureBootSource`. This
+    /// action can only be called before the microVM has booted.
+    ConfigureBootSource(BootSourceConfig),
+    /// Configure the logger using as input the `LoggerConfig`. This action can only be called
+    /// before the microVM has booted.
+    ConfigureLogger(LoggerConfig),
+    /// Get the configuration of the microVM.
+    GetVmConfiguration,
+    /// Flush the metrics. This action can only be called after the logger has been configured.
+    FlushMetrics,
+    /// Add a new block device or update one that already exists using the `BlockDeviceConfig` as
+    /// input. This action can only be called before the microVM has booted.
+    InsertBlockDevice(BlockDeviceConfig),
+    /// Add a new network interface config or update one that already exists using the
+    /// `NetworkInterfaceConfig` as input. This action can only be called before the microVM has
+    /// booted.
+    InsertNetworkDevice(NetworkInterfaceConfig),
+    /// Set the vsock device or update the one that already exists using the
+    /// `VsockDeviceConfig` as input. This action can only be called before the microVM has
+    /// booted.
+    SetVsockDevice(VsockDeviceConfig),
+    /// Update the size of an existing block device specified by an ID. The ID is the first data
+    /// associated with this enum variant. This action can only be called after the microVM is
+    /// started.
+    RescanBlockDevice(String),
```

Comment:
This action was removed in 53cf1bacadc23a21df361a894ac2887cafcb7139. We should also remove it from here.

Reply:
Good catch! This was removed from everywhere else, this is a leftover. Fixed.